okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -398,10 +404,23 @@ public void close() {}
         BufferedSink sink;
         Socket sock;
         try {
-          sock = new Socket(address.getAddress(), address.getPort());
+          sock = new Socket();
+          int timeout = 0; // infinity
+          if (deadline != null) {
+            long timeRemaining = deadline.timeRemaining(TimeUnit.MILLISECONDS);
+            if (timeRemaining <= 0) {
+              timeout = 1000;
```

Comment:
A comment here that this is avoiding the "0 means infinity" would be great. It is a bit strange that `1000` is used given that if `timeRemaining == 3` it stays at 3. Maybe use `1` instead?